File: Application/Models/Order/OrderModel.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Application.Models.Base;
using Application.Models.Customer;
using Core.Enums;

namespace Application.Models.Order
{
    public class OrderModel: BaseModel
    {
        [Required]
        public string CustomerId { get; set; }

        public CustomerModel Customer { get; set; }

        [Required]
        public Status Status { get; set; }

        public decimal? TotalCost { get; set; }

        public string Comment { get; set; }

        public List<OrderItemModel> OrderItems { get; set; }
    }
}

File: Web/Client/src/app/modules/order/resources/models/order.model.ts
import { Customer } from "src/app/modules/customer/resources/models/customer.model"
import { OrderItem } from "./order-item.model";
import { Status } from "./status.enum";

export interface Order{
  id: string
  customerId: string
  customer: Customer
  status: Status
  totalCost: number
  comment: string
  orderItems: OrderItem[]
  createdDate: string
}
